builder/processor/src/main/java/io/helidon/builder/processor/BlueprintProcessor.java

```diff
@@ -0,0 +1,574 @@
+import java.util.LinkedHashSet;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import javax.annotation.processing.AbstractProcessor;
+import javax.annotation.processing.Filer;
+import javax.annotation.processing.Messager;
+import javax.annotation.processing.ProcessingEnvironment;
+import javax.annotation.processing.RoundEnvironment;
+import javax.lang.model.SourceVersion;
+import javax.lang.model.element.Element;
+import javax.lang.model.element.ElementKind;
+import javax.lang.model.element.TypeElement;
+import javax.lang.model.util.Elements;
+import javax.tools.Diagnostic;
+import javax.tools.JavaFileObject;
+
+import io.helidon.builder.processor.ValidationTask.ValidateConfiguredType;
+import io.helidon.common.Errors;
+import io.helidon.common.processor.CopyrightHandler;
+import io.helidon.common.processor.GeneratedAnnotationHandler;
+import io.helidon.common.processor.TypeInfoFactory;
+import io.helidon.common.types.TypeInfo;
+import io.helidon.common.types.TypeName;
+
+import static io.helidon.builder.processor.Types.FACTORY_METHOD;
+import static io.helidon.builder.processor.Types.GENERATED;
+import static io.helidon.builder.processor.Types.OVERRIDE;
+import static io.helidon.builder.processor.Types.PROTOTYPE_BLUEPRINT;
+import static io.helidon.builder.processor.Types.RUNTIME_PROTOTYPE;
+import static io.helidon.builder.processor.Types.RUNTIME_PROTOTYPE_TYPE;
+
+/**
+ * Annotation processor for prototype blueprints.
+ * Generates prototype implementation from the blueprint.
+ */
+public class BlueprintProcessor extends AbstractProcessor {
+    private static final String SOURCE_SPACING = "    ";
+    private static final TypeName GENERATOR = TypeName.create(BlueprintProcessor.class);
+    private final Set<ValidationTask> validationTasks = new LinkedHashSet<>();
+    private final Set<Element> runtimeTypes = new HashSet<>();
+    private final Set<Element> blueprintTypes = new HashSet<>();
+
+    private TypeElement blueprintAnnotationType;
+    private TypeElement runtimePrototypeAnnotationType;
+    private Messager messager;
+    private Filer filer;
+    private ProcessingEnvironment env;
+    private Elements elementUtils;
+
+    @Override
+    public Set<String> getSupportedAnnotationTypes() {
+        return Set.of(PROTOTYPE_BLUEPRINT,
+                      RUNTIME_PROTOTYPE,
+                      GENERATED);
+    }
+
+    @Override
+    public SourceVersion getSupportedSourceVersion() {
+        return SourceVersion.latestSupported();
+    }
+
+    @Override
+    public synchronized void init(ProcessingEnvironment processingEnv) {
+        super.init(processingEnv);
+
+        this.elementUtils = processingEnv.getElementUtils();
+        this.messager = processingEnv.getMessager();
+        this.blueprintAnnotationType = elementUtils.getTypeElement(PROTOTYPE_BLUEPRINT);
+        this.runtimePrototypeAnnotationType = elementUtils.getTypeElement(RUNTIME_PROTOTYPE);
```

Comment:
Objects.requireNonNull(getTypeElement()) for extra level of sanity check.

Reply:
added a null check and a descriptive error message